### PR TITLE
Use wheel unpack instead of unzip

### PR DESCRIPTION
**Pull request by kelkawi-a**
Usage of ```unzip``` requires it to be installed using ```apt-get``` which sometimes faces proxy issues. This change opts for the use of ```wheel unpack``` with some slight modifications to execute the same logic.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -174,12 +174,29 @@ def _process_wheel_file(self, event):  # noqa: C901
...
+                # Rename wheel file to its original name and install it
+                container.exec(["mv", wheel_file, original_wheel_file]).wait()
+
+                _, error = container.exec(["pip", "install", original_wheel_file]).wait_output()
+                if error is not None and error.strip() != "" and not error.strip().startswith("WARNING"):
+                    logger.error(f"failed to install wheel file: {error}")
+                    raise ValueError("Invalid state: failed to install wheel file")
+
+                # Unpack wheel file
+                container.exec(["pip", "install", "wheel"]).wait()
```
Comment (gtato) [inline src/charm.py:187]: can you add this at the worker dependencies ?

File: src/charm.py [hunk 2]
```diff
@@ -174,12 +174,29 @@ def _process_wheel_file(self, event):  # noqa: C901
 
                 # Push wheel file to the container and extract it.
                 container.push(wheel_file, wheel_data, make_dirs=True)
-                container.exec(["apt-get", "update"]).wait()
-                container.exec(["apt-get", "install", "unzip"]).wait()
-                container.exec(["unzip", wheel_file, "-d", "/user_provided"]).wait()
+
```
Comment (gtato) [inline src/charm.py:177]: nothing to do with this line, but maybe you don't have to be in the with block the whole time no?

File: src/charm.py (hunk 1 above)
Comment (kelkawi-a) [reply src/charm.py]: Updated, thanks!

**Review by gtato — APPROVED** [review]
nice

**Pull request merged**